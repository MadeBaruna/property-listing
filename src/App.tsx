import React from 'react';
import { GlobalStyle } from './GlobalStyle';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import Detail from './pages/Detail';
import Search from './pages/Search';

const App: React.FC = () => {
  return (
    <div className="App">
      <GlobalStyle />
      <h1>Property Listing</h1>
      <Router>
        <Switch>
          <Route exact path="/" component={Search} />
          <Route exact path="/property/:id" component={Detail} />
        </Switch>
      </Router>
    </div>
  );
}

export default App;
