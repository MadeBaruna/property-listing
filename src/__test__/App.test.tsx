import React from 'react';
import { render } from 'enzyme';
import 'jest-styled-components'

import App from '../App';

it('renders without crashing', () => {
  const wrapper = render(<App />);
  expect(wrapper).toMatchSnapshot();
});
