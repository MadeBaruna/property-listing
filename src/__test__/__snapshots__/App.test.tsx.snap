// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
.c10 {
  border: none;
  border-radius: 4px;
  padding: 6px 12px;
  line-height: 24px;
  font-size: 16px;
  background: #04ACF4;
  color: white;
  cursor: pointer;
  font-family: 'Segoe UI',Tahoma,Geneva,Verdana,sans-serif;
  -webkit-transition: 200ms ease-out;
  transition: 200ms ease-out;
}

.c10:hover {
  background: #0484f4;
  -webkit-transition: 200ms ease-out;
  transition: 200ms ease-out;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background: white;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: calc(50% - 20px);
  max-width: 320px;
  margin: 10px;
  cursor: pointer;
  -webkit-transition: 100ms ease-out;
  transition: 100ms ease-out;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
}

.c8 img {
  object-fit: cover;
  width: 100%;
}

.c8:hover {
  box-shadow: 0px 13px 21px -5px rgba(0,0,0,0.2);
  -webkit-transition: 100ms ease-out;
  transition: 100ms ease-out;
}

.c9 {
  padding: 10px;
}

.c9 h3 {
  white-space: nowrap;
  overflow: hidden;
  display: block;
  text-overflow: ellipsis;
}

.c9 p {
  height: 100%;
  line-height: 18px;
  max-height: 36px;
  word-wrap: break-word;
  overflow: hidden;
  margin-top: 5px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c9 p.city {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 5px;
}

.c9 svg {
  margin-right: 5px;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: calc(100vh - 70px);
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  padding: 10px;
  width: 50%;
  max-width: 681px;
}

.c11 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c12 {
  height: 100%;
  width: 100%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: white;
  height: 40px;
  margin-left: 20px;
  box-shadow: 0 0 0 2px rgba(0,0,0,.1);
  max-width: 500px;
  width: 100%;
  border-radius: 2px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  border: none;
  padding: 0 10px;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-size: 16px;
  width: 100%;
  background: transparent;
}

.c2:focus {
  outline: none;
}

.c3 {
  margin-right: 5px;
  height: 100%;
  border: none;
  font-size: 16px;
}

.c3:focus {
  outline: none;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: rgba(0,0,0,.1);
  width: 50px;
  height: 100%;
  border: none;
  cursor: pointer;
}

.c4:focus {
  outline: none;
}

.c5 {
  fill: black;
}

.c5.focused {
  fill: white;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: white;
  height: 70px;
  padding: 0px 20px;
  box-shadow: 0px 1px 5px 0px rgba(0,0,0,0.05);
}

<div
  class="App"
>
  <div
    class="c0"
  >
    <h3>
      Property Listing
    </h3>
    <div
      class="c1"
    >
      <input
        class="c2"
      />
      <select
        class="c3"
      >
        <option>
          Apartment
        </option>
        <option>
          Office
        </option>
      </select>
      <button
        class="c4"
      >
        <svg
          class="mdi-icon c5"
          fill="currentColor"
          height="30"
          viewBox="0 0 24 24"
          width="30"
        >
          <path
            d="M9.5,3C13.09,3 16,5.91 16,9.5C16,11.11 15.41,12.59 14.44,13.73L14.71,14H15.5L20.5,19L19,20.5L14,15.5V14.71L13.73,14.44C12.59,15.41 11.11,16 9.5,16C5.91,16 3,13.09 3,9.5C3,5.91 5.91,3 9.5,3M9.5,5C7,5 5,7 5,9.5C5,12 7,14 9.5,14C12,14 14,12 14,9.5C14,7 12,5 9.5,5Z"
          />
        </svg>
      </button>
    </div>
  </div>
  <div
    class="c6"
  >
    <div
      class="c7"
    >
      <div
        class="c8"
      >
        <img
          alt="thumbnail"
          src="https://loremflickr.com/320/240/apartment,office?lock=1"
        />
        <div
          class="c9"
        >
          <h3>
            Mediterania Garden Residence 22222222
          </h3>
          <p>
            Mediterania Garden Residence 2 berlokasi di Podomoro City Jl. Letjen S.Parman Kav.28. Apartemen ini dikembangkan oleh PT Agung Podomoro Land , Tbk,Mediterania Garden Residence 2 memiliki 6 tower. Dalam kompleks ini tersedia beragam fasilitas untuk mendukung hunian Anda, Terdapat pula beragam sarana dan fasilitas yang menjamin kenyamanan hunian Anda, seperti: Access Card, Atm Center, Basketball Court, Bbq Pit, Drug Store, Function Hall, Gym, Jogging Track, Laundry, Mini Market, Movie Theater, Playground, Restaurant, Salon, Super Market, Swimming Pool dan Tennis Court.
          </p>
          <p
            class="city"
          >
            <svg
              class="mdi-icon "
              fill="currentColor"
              height="24"
              viewBox="0 0 24 24"
              width="24"
            >
              <path
                d="M19,15H17V13H19M19,19H17V17H19M13,7H11V5H13M13,11H11V9H13M13,15H11V13H13M13,19H11V17H13M7,11H5V9H7M7,15H5V13H7M7,19H5V17H7M15,11V5L12,2L9,5V7H3V21H21V11H15Z"
              />
            </svg>
             Jakarta Barat
          </p>
          <button
            class="c10"
            style="float:right"
          >
            Lihat Detail
          </button>
        </div>
      </div>
      <div
        class="c8"
      >
        <img
          alt="thumbnail"
          src="https://loremflickr.com/320/240/apartment,office?lock=1"
        />
        <div
          class="c9"
        >
          <h3>
            Mediterania Garden Residence 2
          </h3>
          <p>
            Mediterania Garden Residence 2 berlokasi di Podomoro City Jl. Letjen S.Parman Kav.28. Apartemen ini dikembangkan oleh PT Agung Podomoro Land , Tbk,Mediterania Garden Residence 2 memiliki 6 tower. Dalam kompleks ini tersedia beragam fasilitas untuk mendukung hunian Anda, Terdapat pula beragam sarana dan fasilitas yang menjamin kenyamanan hunian Anda, seperti: Access Card, Atm Center, Basketball Court, Bbq Pit, Drug Store, Function Hall, Gym, Jogging Track, Laundry, Mini Market, Movie Theater, Playground, Restaurant, Salon, Super Market, Swimming Pool dan Tennis Court.
          </p>
          <p
            class="city"
          >
            <svg
              class="mdi-icon "
              fill="currentColor"
              height="24"
              viewBox="0 0 24 24"
              width="24"
            >
              <path
                d="M19,15H17V13H19M19,19H17V17H19M13,7H11V5H13M13,11H11V9H13M13,15H11V13H13M13,19H11V17H13M7,11H5V9H7M7,15H5V13H7M7,19H5V17H7M15,11V5L12,2L9,5V7H3V21H21V11H15Z"
              />
            </svg>
             Jakarta Barat
          </p>
          <button
            class="c10"
            style="float:right"
          >
            Lihat Detail
          </button>
        </div>
      </div>
      <div
        class="c8"
      >
        <img
          alt="thumbnail"
          src="https://loremflickr.com/320/240/apartment,office?lock=1"
        />
        <div
          class="c9"
        >
          <h3>
            Mediterania Garden Residence 2
          </h3>
          <p>
            Mediterania Garden Residence 2 berlokasi di Podomoro City Jl. Letjen S.Parman Kav.28. Apartemen ini dikembangkan oleh PT Agung Podomoro Land , Tbk,Mediterania Garden Residence 2 memiliki 6 tower. Dalam kompleks ini tersedia beragam fasilitas untuk mendukung hunian Anda, Terdapat pula beragam sarana dan fasilitas yang menjamin kenyamanan hunian Anda, seperti: Access Card, Atm Center, Basketball Court, Bbq Pit, Drug Store, Function Hall, Gym, Jogging Track, Laundry, Mini Market, Movie Theater, Playground, Restaurant, Salon, Super Market, Swimming Pool dan Tennis Court.
          </p>
          <p
            class="city"
          >
            <svg
              class="mdi-icon "
              fill="currentColor"
              height="24"
              viewBox="0 0 24 24"
              width="24"
            >
              <path
                d="M19,15H17V13H19M19,19H17V17H19M13,7H11V5H13M13,11H11V9H13M13,15H11V13H13M13,19H11V17H13M7,11H5V9H7M7,15H5V13H7M7,19H5V17H7M15,11V5L12,2L9,5V7H3V21H21V11H15Z"
              />
            </svg>
             Jakarta Barat
          </p>
          <button
            class="c10"
            style="float:right"
          >
            Lihat Detail
          </button>
        </div>
      </div>
      <div
        class="c8"
      >
        <img
          alt="thumbnail"
          src="https://loremflickr.com/320/240/apartment,office?lock=1"
        />
        <div
          class="c9"
        >
          <h3>
            Mediterania Garden Residence 2
          </h3>
          <p>
            Mediterania Garden Residence 2 berlokasi di Podomoro City Jl. Letjen S.Parman Kav.28. Apartemen ini dikembangkan oleh PT Agung Podomoro Land , Tbk,Mediterania Garden Residence 2 memiliki 6 tower. Dalam kompleks ini tersedia beragam fasilitas untuk mendukung hunian Anda, Terdapat pula beragam sarana dan fasilitas yang menjamin kenyamanan hunian Anda, seperti: Access Card, Atm Center, Basketball Court, Bbq Pit, Drug Store, Function Hall, Gym, Jogging Track, Laundry, Mini Market, Movie Theater, Playground, Restaurant, Salon, Super Market, Swimming Pool dan Tennis Court.
          </p>
          <p
            class="city"
          >
            <svg
              class="mdi-icon "
              fill="currentColor"
              height="24"
              viewBox="0 0 24 24"
              width="24"
            >
              <path
                d="M19,15H17V13H19M19,19H17V17H19M13,7H11V5H13M13,11H11V9H13M13,15H11V13H13M13,19H11V17H13M7,11H5V9H7M7,15H5V13H7M7,19H5V17H7M15,11V5L12,2L9,5V7H3V21H21V11H15Z"
              />
            </svg>
             Jakarta Barat
          </p>
          <button
            class="c10"
            style="float:right"
          >
            Lihat Detail
          </button>
        </div>
      </div>
    </div>
    <div
      class="c11"
    >
      <div
        class="c12"
      />
    </div>
  </div>
</div>
`;
